s3same/iam.py

```diff
@@ -0,0 +1,75 @@
+from botocore.exceptions import ClientError
+
+IAMName = 's3same_travis'
+
+def _policy_string(bucket):
+    return """{
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Action": [
+            "s3:ListBucket"
+          ],
+          "Effect": "Allow",
+          "Resource": [
+            "arn:aws:s3:::%(bucket)s"
+          ]
+        },
+        {
+          "Action": [
+            "s3:PutObject",
+            "s3:PutObjectAcl"
+          ],
+          "Effect": "Allow",
+          "Resource": [
+            "arn:aws:s3:::%(bucket)s/*"
+          ]
+        }
+      ]
+    }""" % dict(bucket=bucket)
+
+def _policy_arn(iam, bucket):
+    policy_arn = None
+    try:
+        response = iam.create_policy(
+                PolicyName=IAMName,
+                PolicyDocument=_policy_string(bucket),
+                )
+        policy_arn = response.get('Policy', {}).get('Arn')
+    except ClientError as e:
+        if e.response['Error']['Code'] != 'EntityAlreadyExists':
+            raise e
+        marker = None
+        while not policy_arn:
+            response = iam.list_policies(Scope='Local')
+            policy_arn = next((p.get('Arn') for p in response.get('Policies', []) if p.get('PolicyName') == IAMName), None)
+            if not response.get('IsTruncated'):
+                break
+            marker = response['Marker']
```

Comment:
Yeah, sorry. If you can't tell by the 30 minute pauses between comments, my review of this has been somewhat interrupted, and I lost the thread a bit. lol You're right, I think keeping it inside the except clause with the `while True` is a good way to go, though one of the other guys may disagree? And definitely agreed re: generator expression.